.circleci/config.yml

```diff
@@ -38,177 +38,232 @@ defaults: &defaults
+            - vendor-{{ checksum "composer.lock" }}
+            - vendor-
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: /build-tools-ci/scripts/set-environment
+
+      - run:
+          name: run PHP build step
+          command: ./.ci/build/php
+
+      - save_cache:
+          key: vendor-{{ checksum "composer.lock" }}
+          paths:
+            - $CIRCLE_WORKING_DIRECTORY/vendor
+
+      - persist_to_workspace:
+          root: .
+          paths:
+            - web
+            - vendor
+            - drush
+
+  deploy_to_pantheon:
+    <<: *defaults
+    steps:
+      - checkout
+
+      # Attach the workspace
+      - attach_workspace:
+          at: /tmp
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: /build-tools-ci/scripts/set-environment
+
+      # Rsync the built web directory
+      - run:
+          name: sync web
+          command: rsync -az /tmp/web/ web
+
+      # Rsync the built vendor directory
+      - run:
+          name: sync vendor
+          command: rsync -az /tmp/vendor/ vendor
+
+      # Rsync the drush directory
+      - run:
+          name: sync drush
+          command: rsync -az /tmp/drush/ drush
+
+      # Deploy to Pantheon
+      - run:
+          name: deploy to Pantheon
+          command: ./.ci/deploy/pantheon/dev-multidev
+
+  behat_test:
+    <<: *defaults
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - composer-cache-{{ checksum "composer.lock" }}
+            - composer-cache-
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: /build-tools-ci/scripts/set-environment
+
+      - run:
+          name: prepare for Behat testing
+          command: ./.ci/test/behat/initialize
+
+      - run:
+          name: run acceptance tests with Behat
+          command: ./.ci/test/behat/run
+
+      - run:
+          name: clean up after Behat
+          command: ./.ci/test/behat/cleanup
+          when: always
+
+      - save_cache:
+          key: composer-cache-{{ checksum "composer.lock" }}
+          paths:
+            - $HOME/.composer/cache
+
+      - store_artifacts:
+          path: /tmp/artifacts
+          destination: artifacts
+
+  visual_regression_test:
+    <<: *defaults
+    docker:
+      - image: backstopjs/backstopjs:4.1.9
+    steps:
+      - checkout
+
+      - attach_workspace:
+          at: /tmp/workspace
+
+      - run: cp /tmp/workspace/bash_env.txt $BASH_ENV
```

Comment:
I guess the reason this is the only step where bash_env is restored is that this container cannot run the set-environment script.

Reply:
That is correct

Reply:
An alternative could be to persist the `/build-tools-ci/scripts/set-environment`, rather than the resulting `$BASH_ENV` file, and run the `set-environment` script directly.